src/forms/auth/loginView.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from "react";
+import { RouteComponentProps, withRouter } from "react-router";
+import WrappedNormalLoginForm from "./login";
+import { Classes, AnchorButton } from "@blueprintjs/core";
+import { connect } from "react-redux";
+import "./loginView.scss";
+import * as QueryString from "query-string"
+import { isNullOrUndefined } from "util";
+import axios from "axios";
+import { API_ROOT } from "../../utils/api-config";
+import * as actions from "../../store/actions/auth";
+
+export interface LoginViewProps {
+    token: string;
+    onAuth(acess_token: string): any;
+}
+export class LoginView extends React.PureComponent<RouteComponentProps & LoginViewProps> {
+    private handleSSOClick = () => {
+        const redirect_uri = window.location.href.replace(new RegExp(':', 'g'), '%3A').replace(new RegExp('/', 'g'), '%2F')
```

Comment:
can't have any `/` or `:` characters since this is a parameter in the URL

Reply:
regex :o